.github/workflows/cont_integration.yml

```diff
@@ -32,14 +32,16 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@v2
+      - name: Generate cache key
+        run: echo "${{ matrix.rust }} ${{ matrix.features }}" | tee .cache_key
       - name: cache
         uses: actions/cache@v2
         with:
           path: |
             ~/.cargo/registry
             ~/.cargo/git
             target
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml','**/Cargo.lock') }}
+          key: ${{ runner.os }}-cargo-${{ hashFiles('.cache_key') }}-${{ hashFiles('**/Cargo.toml','**/Cargo.lock') }}
```

Comment:
I think we can get the same level of cache isolation without creating the `.cache_key` file by using:
```
key: ${{ runner.os }}-cargo-${{ matrix.rust }}-${{ matrix.features }}-${{ hashFiles('**/Cargo.toml','**/Cargo.lock') }}
```
not a big deal, but if you agree I'll make a PR for this change.

Reply:
I was having issues with that approach because the key cannot contain any commas, and the `matrix.features` variable most of the times had some in it

Reply:
ah ok makes sense to do it the way you did it then.